stdlib/public/core/Builtin.swift

```diff
@@ -158,6 +158,7 @@ internal func != (lhs: Builtin.RawPointer, rhs: Builtin.RawPointer) -> Bool {
   return !(lhs == rhs)
 }
 
+#if compiler(>=5.3) && $GeneralizedIsSameMetaTypeBuiltin
```

Comment:
Surrounding the declaration with an `#if $GeneralizedIsSameMetaTypeBuiltin` check won't cause the `#if $GeneralizedIsSameMetaTypeBuiltin` to appear in the swiftinterface. It's only the `#if`s in function bodies that are reproduced verbatim. The AST printer can be taught to emit `#if $GeneralizedIsSameMetaTypeBuiltin` around declarations automatically if there's a way for the compiler to detect that the decl needs it.